e2e/nomostest/config_sync.go

```diff
@@ -974,20 +1007,34 @@ func (nt *NT) RootSyncObjectOCI(name string, image *registryproviders.OCIImage)
 	default:
 		nt.T.Fatalf("Unrecognized OCIProvider: %s", *e2e.OCIProvider)
 	}
+	if nt.DefaultReconcileTimeout != nil {
+		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*nt.DefaultReconcileTimeout)
+	} else if rs.Spec.Override != nil {
+		rs.Spec.Override.ReconcileTimeout = nil
+	}
+	// Enable automatic deletion of managed objects by default.
+	// This helps ensure that test artifacts are cleaned up.
+	EnableDeletionPropagation(rs)
 	return rs
 }
 
 // RepoSyncObjectOCI returns a RepoSync object that syncs the provided OCIImage.
-func (nt *NT) RepoSyncObjectOCI(nn types.NamespacedName, image *registryproviders.OCIImage) *v1beta1.RepoSync {
-	imageURL, err := image.RemoteAddressWithTag()
-	if err != nil {
-		nt.T.Fatalf("OCIImage.RemoteAddressWithTag: %v", err)
+// ImageID is used for pulling. ImageDigest is used for validating the result.
+// Creates, updates, or replaces the SyncSource expectation for this RepoSync.
+func (nt *NT) RepoSyncObjectOCI(nn types.NamespacedName, imageID registryproviders.OCIImageID, syncPath, expectedImageDigest string) *v1beta1.RepoSync {
+	id := core.RepoSyncID(nn.Name, nn.Namespace)
+	// Register the OCI image for syncing
+	expectedSyncPath := syncPath
+	if syncPath == "" {
+		expectedSyncPath = controllers.DefaultSyncDir
 	}
-
-	rs := RepoSyncObjectV1Beta1FromNonRootRepo(nt, nn)
+	SetExpectedOCISource(nt, id, imageID, expectedImageDigest, expectedSyncPath)
+	rs := k8sobjects.RepoSyncObjectV1Beta1(nn.Namespace, nn.Name)
+	rs.Spec.SourceFormat = configsync.SourceFormatUnstructured // TODO: is SourceFormat necessary with OCI?
```

Comment:
RepoSync always uses unstructured format.

Reply:
We don't need to specify it for RepoSyncs, right?

Reply:
Correct. The default value is unstructured for RepoSyncs.